### PR TITLE
Replace Microsoft in GCM namespaces/file names

### PR DESCRIPTION
**Pull request by ldennington**
Overview
----------------------
With this change, we remove `Microsoft` from GCM Core's project, directory, namespace, and assembly names. We made these updates in a series of four commits corresponding to the following:

1. Update/set explicit assembly names in .csproj files (and update XAML resource/namespace paths to reflect)
2. Update/set explicit namespaces
3. Rename projects/directories
4. Sign `gcmcore.dll`

### CODE REVIEW

**Review by mjcheetham — CHANGES_REQUESTED** [review]
@ldennington thanks for this! I've noticed a problem however 😢
Running the GitHub.UI and Atlassian.Bitbucket.UI helpers results in a crash in failing to resolve XAML resources(!)

I think the problem is that when updating the [a]xaml files, resource paths need the _assembly name_, and not the namespace. In WPF it is `pack://application:,,,/<assembly>;component/<path>` and in Avalonia it is `avares://<assembly>/<path>`.

Here's a patch that should fix the missing/problematic resource paths. It should be applied after the first commit (and then there'll need to be some conflict resolution on the 2nd commit):

```diff
diff --git a/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Assets/Controls.axaml b/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Assets/Controls.axaml
index c1f62b4..9f10e47 100644
--- a/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Assets/Controls.axaml
+++ b/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Assets/Controls.axaml
@@ -1,6 +1,6 @@
 <Styles xmlns="https://github.com/avaloniaui"
         xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
-    <StyleInclude Source="avares://Microsoft.Git.CredentialManager.UI.Avalonia/Assets/ButtonHyperlink.axaml"/>
+    <StyleInclude Source="avares://gcmcoreuiavn/Assets/ButtonHyperlink.axaml"/>
     <Style Selector="TextBox.monospace">
         <Setter Property="FontFamily" Value="SF Mono, Courier New, Courier, monospace" />
     </Style>
diff --git a/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/AvaloniaApp.axaml b/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/AvaloniaApp.axaml
index 849e63b..486f84c 100644
--- a/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/AvaloniaApp.axaml
+++ b/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/AvaloniaApp.axaml
@@ -4,12 +4,12 @@
              Name="Git Credential Manager">
     <Application.Styles>
         <FluentTheme Mode="Light"/>
-        <StyleInclude Source="avares://Microsoft.Git.CredentialManager.UI.Avalonia/Assets/Controls.axaml"/>
+        <StyleInclude Source="avares://gcmcoreuiavn/Assets/Controls.axaml"/>
     </Application.Styles>
     <Application.Resources>
         <ResourceDictionary>
             <ResourceDictionary.MergedDictionaries>
-                <ResourceInclude Source="avares://Microsoft.Git.CredentialManager.UI.Avalonia/Assets/LightBase.axaml"/>
+                <ResourceInclude Source="avares://gcmcoreuiavn/Assets/LightBase.axaml"/>
             </ResourceDictionary.MergedDictionaries>
         </ResourceDictionary>
     </Application.Resources>
diff --git a/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Controls/AboutWindow.axaml b/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Controls/AboutWindow.axaml
index 22ec73b..7658a77 100644
--- a/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Controls/AboutWindow.axaml
+++ b/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Controls/AboutWindow.axaml
@@ -10,12 +10,12 @@
     <Window.Resources>
         <ResourceDictionary>
             <ResourceDictionary.MergedDictionaries>
-                <ResourceInclude Source="avares://Microsoft.Git.CredentialManager.UI.Avalonia/Assets/Images.axaml"/>
+                <ResourceInclude Source="gcmcoreuiavn/Assets/Images.axaml"/>
             </ResourceDictionary.MergedDictionaries>
         </ResourceDictionary>
     </Window.Resources>
     <Window.Styles>
-        <StyleInclude Source="avares://Microsoft.Git.CredentialManager.UI.Avalonia/Assets/Controls.axaml"/>
+        <StyleInclude Source="avares://gcmcoreuiavn/Assets/Controls.axaml"/>
     </Window.Styles>
     <StackPanel Margin="20">
         <Image HorizontalAlignment="Center"
diff --git a/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Controls/DialogWindow.axaml b/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Controls/DialogWindow.axaml
index 2940c8d..e0949cf 100644
--- a/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Controls/DialogWindow.axaml
+++ b/src/shared/Microsoft.Git.CredentialManager.UI.Avalonia/Controls/DialogWindow.axaml
@@ -2,7 +2,7 @@
         xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
         xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
         xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
-        xmlns:vm="clr-namespace:Microsoft.Git.CredentialManager.UI.ViewModels;assembly=Microsoft.Git.CredentialManager.UI"
+        xmlns:vm="clr-namespace:Microsoft.Git.CredentialManager.UI.ViewModels;assembly=gcmcoreui"
         xmlns:converters="clr-namespace:Microsoft.Git.CredentialManager.UI.Converters"
         mc:Ignorable="d" d:DesignWidth="420" d:DesignHeight="520"
         x:Class="Microsoft.Git.CredentialManager.UI.Controls.DialogWindow"
diff --git a/src/windows/Atlassian.Bitbucket.UI.Windows/Assets/Styles.xaml b/src/windows/Atlassian.Bitbucket.UI.Windows/Assets/Styles.xaml
index 338e98e..67fe8a5 100644
--- a/src/windows/Atlassian.Bitbucket.UI.Windows/Assets/Styles.xaml
+++ b/src/windows/Atlassian.Bitbucket.UI.Windows/Assets/Styles.xaml
@@ -1,6 +1,6 @@
 ﻿<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
                     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
     <ResourceDictionary.MergedDictionaries>
-        <ResourceDictionary Source="pack://application:,,,/Microsoft.Git.CredentialManager.UI.Windows;component/Assets/Styles.xaml"/>
+        <ResourceDictionary Source="pack://application:,,,/gcmcoreuiwpf;component/Assets/Styles.xaml"/>
     </ResourceDictionary.MergedDictionaries>
 </ResourceDictionary>
diff --git a/src/windows/GitHub.UI.Windows/Assets/Styles.xaml b/src/windows/GitHub.UI.Windows/Assets/Styles.xaml
index 338e98e..67fe8a5 100644
--- a/src/windows/GitHub.UI.Windows/Assets/Styles.xaml
+++ b/src/windows/GitHub.UI.Windows/Assets/Styles.xaml
@@ -1,6 +1,6 @@
 ﻿<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
                     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
     <ResourceDictionary.MergedDictionaries>
-        <ResourceDictionary Source="pack://application:,,,/Microsoft.Git.CredentialManager.UI.Windows;component/Assets/Styles.xaml"/>
+        <ResourceDictionary Source="pack://application:,,,/gcmcoreuiwpf;component/Assets/Styles.xaml"/>
     </ResourceDictionary.MergedDictionaries>
 </ResourceDictionary>
diff --git a/src/windows/Shared.UI.Windows/Controls/DialogWindow.xaml b/src/windows/Shared.UI.Windows/Controls/DialogWindow.xaml
index ebdf112..ef262d5 100644
--- a/src/windows/Shared.UI.Windows/Controls/DialogWindow.xaml
+++ b/src/windows/Shared.UI.Windows/Controls/DialogWindow.xaml
@@ -3,7 +3,7 @@
         xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
         xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
         xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" mc:Ignorable="d"
-        xmlns:viewModels="clr-namespace:Microsoft.Git.CredentialManager.UI.ViewModels;assembly=Microsoft.Git.CredentialManager.UI"
+        xmlns:viewModels="clr-namespace:Microsoft.Git.CredentialManager.UI.ViewModels;assembly=gcmcoreui"
         Width="414"
         SizeToContent="Height"
         MinHeight="320"

```

File: src/windows/Payload.Windows/Payload.Windows.csproj [hunk 1]
```diff
@@ -28,8 +28,8 @@
       $(OutDir)GitHub.dll;
       $(OutDir)GitHub.UI.exe;
       $(OutDir)Microsoft.AzureRepos.dll;
-      $(OutDir)Microsoft.Git.CredentialManager.dll;
-      $(OutDir)Microsoft.Git.CredentialManager.UI.dll;">
```
Comment (mjcheetham) [inline src/windows/Payload.Windows/Payload.Windows.csproj:32]: This is not a problem with the changes here, but looks like we've not been signing the `gcmcoreui.dll` assembly (to use the new names)! Oops..

File: Git-Credential-Manager.sln [hunk 2]
```diff
@@ -37,7 +37,7 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Atlassian.Bitbucket", "src\
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Atlassian.Bitbucket.Tests", "src\shared\Atlassian.Bitbucket.Tests\Atlassian.Bitbucket.Tests.csproj", "{025E5329-A0B1-4BA9-9203-B70B44A5F9E0}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Shared.UI.Windows", "src\windows\Shared.UI.Windows\Shared.UI.Windows.csproj", "{2B3CD8FF-84A6-4B53-A28B-D7A75B0AB4D7}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "GitCredentialManager", "src\windows\Shared.UI.Windows\Shared.UI.Windows.csproj", "{2B3CD8FF-84A6-4B53-A28B-D7A75B0AB4D7}"
```
Comment (mjcheetham) [inline Git-Credential-Manager.sln]: This solution file change looks suspicious.. the project file `Shared.UI.Windows.csproj` was not renamed in the 2nd commit, but the name of the project in the solution file seems to have been??

**Review by mjcheetham — COMMENTED** [review]
There's some failing builds it seems with the Windows/WPF UI helpers?
https://github.com/microsoft/Git-Credential-Manager-Core/runs/4069838162?check_suite_focus=true

There's also some suspicious solution file change on the `Shared.UI.Windows` project. The rest looks ok.

File: src/windows/Payload.Windows/Payload.Windows.csproj (hunk 1 above)
Comment (ldennington) [reply src/windows/Payload.Windows/Payload.Windows.csproj:32]: I believe this is fixed now!

File: Git-Credential-Manager.sln (hunk 2 above)
Comment (ldennington) [reply Git-Credential-Manager.sln]: I believe this is fixed now!

Comment (ldennington) [reply Git-Credential-Manager.sln]: Woof but now there is a test issue. Standby.

**Review by mjcheetham — APPROVED** [review]
Looks good to me! These sorts of changes are a right PITA, good job!

**Pull request merged**